fingertip/time_desync.py

```diff
@@ -0,0 +1,48 @@
+# Licensed under GNU General Public License v3 or later, see COPYING.
+# Copyright (c) 2019 Red Hat, Inc., see CONTRIBUTORS.
+
+from fingertip.util import log
+
+
+class TimeDesync:
+    NONE = 0
+    SMALL = 1  # less then execution length: pauses or rewinds
+    LARGE = 2  # less than expiration time: old persisted snapshot was booted
+
+    @classmethod
+    def _parse_scale(cls, scale):
+        if isinstance(scale, str):
+            return getattr(cls, scale.upper())
+        return int(scale)
+
+    def __init__(self, m):
+        self._m = m
+        self._current_scale = TimeDesync.NONE
+        self._allowed_scale = TimeDesync.LARGE
+
+    def report(self, scale):
+        scale = self._parse_scale(scale)
+        if scale > self._current_scale:
+            log.debug(f'time desync scale increased to {scale}')
+            self._current_scale = scale
+
+    def tighten(self, allowed_scale):
```

Comment:
Is this used somewhere?

Reply:
Not yet.

Reply:
For testing, I invoked it directly with `... + .time_desync.tighten NONE + ... `

Reply:
I always forget about this use :) Makes sense. Otherwise it looks sensible.